src/core/tests/preprocess.cpp

```diff
@@ -914,6 +914,39 @@ TEST(pre_post_process, mean_vector_dynamic_channels_shape) {
     EXPECT_EQ(f->get_output_element_type(0), element::f32);
 }
 
+TEST(pre_post_process, pad_vector_constant_layout) {
+    auto f = create_simple_function(element::f32, Shape{1, 3, 199, 199});
+    auto p = PrePostProcessor(f);
+
+    std::stringstream exp_dump;
+    exp_dump << PartialShape{1, 3, 200, 200};
+    try{
+        p.input().preprocess().pad({0, 0, 0, 0}, {0, 0, 1, 1}, 0, PaddingMode::CONSTANT);
+        p.build();
+    } catch (const ov::Exception& err) {
+        // Padded shape should be {1, 3, 200, 200}
+        EXPECT_TRUE(std::string(err.what()).find(exp_dump.str())) << err.what();
```

Comment:
It is not good idea to check pad result in exception message. 
We can combine input `set_shape()` with `pad()`to test it:
1. create single function with shape(1,3,200,200)
2. set input shape(1,3,199,199): `p.input().tensor().set_shape({1, 3, 199, 199});`
3. set `pad()` arguments with `pad_begin={0, 0, 0, 0}, pad_end={0, 0, 1, 1}`
4. check no Exception and pad output shapes